src/cli/bootupctl.rs

```diff
@@ -129,7 +130,14 @@ impl CtlCommand {
     /// Runner for `update` verb.
     fn run_update() -> Result<()> {
         ensure_running_in_systemd()?;
-        bootupd::client_run_update()
+        let sysroot = openat::Dir::open("/").context("Opening root dir")?;
+        let dest_fd = format!("/proc/self/fd/{}", sysroot.as_raw_fd());
+        let dest_root = std::fs::read_link(dest_fd)?;
+
+        let devices = crate::blockdev::get_devices(&dest_root)
+            .with_context(|| "while looking for parent devices")?;
+        crate::blockdev::init_parent_devices(devices);
```

Comment:
Here's where I'd suggest moving the `static` instead.

However even then *in theory* the parent devices are scoped to a particular destination root and so it can't actually be static. Which means it's hard to cache in that way.

So if we wanted to avoid querying the devices multiple times, then what we'd need to do is query them up front and then pass them down. That may involve more code churn though.